### PR TITLE
Edit general information survey metadata and other fixes/changes

### PR DESCRIPTION
**Pull request by sdevalapurkar**
# Overview

## Links to jira tickets

- https://quartech.atlassian.net/browse/BHBC-623

## This PR contains the following changes

- Edit general information survey metadata and other fixes/changes

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [x] Enhancement (improvements to existing functionality)
- [x] Bug fix (change which fixes an issue)

## How Has This Been Tested?

- Locally

## Screenshots

image

image

### CODE REVIEW

File: api/src/models/survey-view-update.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+const defaultLog = getLogger('models/survey-view-update');
+
+/**
+ * Pre-processes GET /project/{projectId}/survey/{surveyId} survey data
+ *
+ * @export
+ * @class GetSurveyData
+ */
+export class GetSurveyData {
```
Comment (anissa-agahchen) [inline api/src/models/survey-view-update.ts:11]: shouldn't this be in /models/survey-view?

Comment (sdevalapurkar) [reply api/src/models/survey-view-update.ts:11]: its for both view and update 😄

Comment (sdevalapurkar) [reply api/src/models/survey-view-update.ts:11]: thats why it is in `view-update`

File: api/src/paths/project/{projectId}/surveys.ts [hunk 2]
```diff
@@ -59,10 +69,14 @@ GET.apiDoc = {
...
   return async (req, res) => {
+    if (!req.params.projectId) {
+      throw new HTTP400('Missing required path param `projectId`');
+    }
+
     const connection = getDBConnection(req['keycloak_token']);
 
     try {
-      const getSurveyListSQLStatement = getSurveyListSQL();
+      const getSurveyListSQLStatement = getSurveyListSQL(Number(req.params.projectId));
```
Comment (anissa-agahchen) [inline api/src/paths/project/{projectId}/surveys.ts:79]: nice

File: app/src/features/surveys/CreateSurveyPage.test.tsx [hunk 3]
```diff
@@ -95,7 +95,7 @@ describe('CreateSurveyPage', () => {
         expect(getByText('Are you sure you want to cancel?')).toBeInTheDocument();
       });
 
-      fireEvent.click(getAllByText('Yes')[2]);
+      fireEvent.click(getAllByText('Yes')[1]);
```
Comment (anissa-agahchen) [inline app/src/features/surveys/CreateSurveyPage.test.tsx]: better with test-id?

Comment (sdevalapurkar) [reply app/src/features/surveys/CreateSurveyPage.test.tsx]: yes, will commit

File: app/src/features/surveys/view/components/SurveyProprietaryData.tsx [hunk 4]
```diff
@@ -35,56 +35,47 @@ const SurveyProprietaryData: React.FC<ISurveyProprietaryDataProps> = (props) =>
...
             <Grid item>
               <Box mt={1}>
                 <Box display="flex" alignItems="center" justifyContent="space-between" height="2rem">
                   <Typography component="dt" variant="subtitle2" color="textSecondary">
-                    Proprietary Information
+                    Category Rational
                   </Typography>
                 </Box>
-                <Typography>The data captured in this survey is not proprietary.</Typography>
+                <Typography style={{ wordBreak: 'break-all' }}>{surveyProprietor.category_rational}</Typography>
```
Comment (anissa-agahchen) [inline app/src/features/surveys/view/components/SurveyProprietaryData.tsx]: spelling: rationale

Comment (sdevalapurkar) [reply app/src/features/surveys/view/components/SurveyProprietaryData.tsx]: 😆

**Review by anissa-agahchen — APPROVED** [review]
Looks good !

**Pull request merged**